src/main/java/core/basesyntax/service/parser/FruitRecordDtoParserImpl.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.service.parser;
+
+import core.basesyntax.model.dto.FruitRecordDto;
+import core.basesyntax.service.Operation;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitRecordDtoParserImpl implements FruitRecordDtoParser {
+    private static final String NEGATIVE_VALUE_EXCEPTION = "Quantity can't be less than 0";
+    private static final String INSUFFICIENT_NUMBER_OF_VALUES = "Current line has less "
+            + "than three elements";
+    private static final String SPLITTER = ",";
+    private static final int OPERATION_TYPE = 0;
+    private static final int FRUIT_NAME = 1;
+    private static final int QUANTITY = 2;
+
+    @Override
+    public List<FruitRecordDto> parse(List<String> lines) {
+        String operationType;
+        String fruitName;
+        int quantity = 0;
+        String[] currentLineArray;
+        List<FruitRecordDto> recordDtos = new ArrayList<>(lines.size());
+        for (String currentLine : lines) {
+            currentLineArray = currentLine.split(SPLITTER);
+            if (currentLineArray.length < 3) {
+                throw new RuntimeException(INSUFFICIENT_NUMBER_OF_VALUES);
+            }
+            operationType = Operation.getOperationByLetter(currentLineArray[OPERATION_TYPE]
+                    .trim()).getOperation();
```

Comment:
why do you trim here? You data without spaces

Reply:
what if I have something like this
.....b,banana,20
I'll get an exception (Invalid operation format,    b)

Reply:
It is not your job. It is the wrong data that you were provided. Just throw an exception. What you will do if you get kkl20hf as amount?

Reply:
Your job is to work with valid csv data all the rest is exception. You don't know how to work with them.